### PR TITLE
[Wasm] Add cluster metadata fallback and upstream host metadata

### PR DESCRIPTION
**Pull request by bianpengyuan**
Signed-off-by: Pengyuan Bian <bianpengyuan@google.com>

cc @kyessenov 

For an explanation of how to fill out the fields, please see the relevant section
in [PULL_REQUESTS.md](https://github.com/envoyproxy/envoy/blob/master/PULL_REQUESTS.md)

Commit Message:
Additional Description:
Risk Level:
Testing:
Docs Changes:
Release Notes:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Deprecated:]

### CODE REVIEW

**Comment by mattklein123** [comment]
Please add tests, thank you!

/wait

File: source/extensions/common/wasm/context.cc [hunk 1]
```diff
@@ -524,6 +525,14 @@ Context::findValue(absl::string_view name, Protobuf::Arena* arena, bool last) co
   case PropertyToken::CLUSTER_METADATA:
     if (info && info->upstreamHost()) {
       return CelValue::CreateMessage(&info->upstreamHost()->cluster().metadata(), arena);
+    } else if (info && info->upstreamClusterInfo().has_value() &&
+               info->upstreamClusterInfo().value()) {
+      return CelValue::CreateMessage(&info->upstreamClusterInfo().value()->metadata(), arena);
+    }
```
Comment (mattklein123) [inline source/extensions/common/wasm/context.cc:531]: Does this have a test?

/wait

Comment (bianpengyuan) [reply source/extensions/common/wasm/context.cc:531]: No seems like we don't have coverage on cluster metadata. Will add test for both conditions.

Comment (bianpengyuan) [reply source/extensions/common/wasm/context.cc:531]: Done.

**Review by mattklein123 — APPROVED** [review]
Thanks!

**Pull request merged**